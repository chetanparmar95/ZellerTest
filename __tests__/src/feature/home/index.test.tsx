import 'react-native';
import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import renderer from 'react-test-renderer';
import Home from '../../../../src/feature/home';
import UserType from '../../../../src/components/userType';

const mockCustomers = [
  {
    id: '1',
    name: 'TestCustomer1',
    email: 'test1@test.com',
    role: 'Admin',
  },
  {
    id: '2',
    name: 'TestCustomer2',
    email: 'test2@test.com',
    role: 'Admin',
  },
];

jest.mock('react-redux', () => {
  const store = {
    user: {customers: {items: mockCustomers}},
  };
  return {
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn().mockImplementation(callback => callback(store)),
    useDispatch: () => jest.fn(),
  };
});

describe('Test Home', () => {
  let wrapper: ShallowWrapper;
  const mockSetSelectedType = jest.fn();

  it('renders correctly', () => {
    React.useState = jest.fn().mockReturnValue(['Admin', mockSetSelectedType]);
    jest.spyOn(React, 'useEffect').mockImplementation(f => f());
    wrapper = shallow(<Home />);
    expect(renderer.create(<Home />)).toMatchSnapshot();
  });

  it('should flatlist return keyExtractor correctly for user types', () => {
    const key = wrapper.find('FlatList').at(0).props().keyExtractor('Admin');
    expect(key).toEqual('Admin');
  });

  it('should flatlist return keyExtractor correctly for user list', () => {
    const key = wrapper.find('FlatList').at(1).props().keyExtractor({id: 1});
    expect(key).toEqual('1');
  });

  it('should UserType render and setSelected called with given value', () => {
    const RenderItem: Function = wrapper
      .find('FlatList')
      .at(0)
      .prop('renderItem');
    const renderItemShallowWrapper = shallow(<RenderItem item={'Admin'} />);
    expect(renderItemShallowWrapper).toMatchSnapshot();
    renderItemShallowWrapper.find(UserType).props().onSelected('Admin');
    expect(mockSetSelectedType).toHaveBeenCalledWith('Admin');
  });

  it('should UserView render', () => {
    const RenderItem: Function = wrapper
      .find('FlatList')
      .at(1)
      .prop('renderItem');
    const renderItemShallowWrapper = shallow(
      <RenderItem item={mockCustomers[0]} />,
    );
    expect(renderItemShallowWrapper).toMatchSnapshot();
  });
});
