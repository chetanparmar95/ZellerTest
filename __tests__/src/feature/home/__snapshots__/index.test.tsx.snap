// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home renders correctly 1`] = `
<View
  style={
    {
      "padding": 16,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "rgb(234,242,250)",
        "height": 1,
        "marginVertical": 16,
        "width": "100%",
      }
    }
  />
  <Text
    style={
      {
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    User Types
  </Text>
  <RCTScrollView
    data={
      [
        "Admin",
        "Manager",
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    style={
      {
        "marginTop": 16,
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgb(234,242,250)",
              "borderRadius": 10,
              "flex": 1,
              "flexDirection": "row",
              "marginVertical": 4,
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderColor": "rgb(48,111,199)",
                  "borderRadius": 12,
                  "borderWidth": 1,
                  "height": 24,
                  "justifyContent": "center",
                  "width": 24,
                },
                {
                  "borderColor": "rgb(48,111,199)",
                },
              ]
            }
          >
            <View
              style={
                {
                  "backgroundColor": "rgb(48,111,199)",
                  "borderRadius": 8,
                  "height": 16,
                  "width": 16,
                }
              }
            />
          </View>
          <Text
            style={
              {
                "marginHorizontal": 16,
              }
            }
          >
            Admin
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 10,
              "flex": 1,
              "flexDirection": "row",
              "marginVertical": 4,
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderColor": "rgb(48,111,199)",
                  "borderRadius": 12,
                  "borderWidth": 1,
                  "height": 24,
                  "justifyContent": "center",
                  "width": 24,
                },
                {
                  "borderColor": "rgb(234,242,250)",
                },
              ]
            }
          />
          <Text
            style={
              {
                "marginHorizontal": 16,
              }
            }
          >
            Manager
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "backgroundColor": "rgb(234,242,250)",
        "height": 1,
        "marginVertical": 16,
        "width": "100%",
      }
    }
  />
  <Text
    style={
      {
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Admin
     
    Users
  </Text>
  <RCTScrollView
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    style={
      {
        "marginTop": 16,
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
  <View
    style={
      {
        "backgroundColor": "rgb(234,242,250)",
        "height": 1,
        "marginVertical": 16,
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`Test Home should UserType render and setSelected called with given value 1`] = `ShallowWrapper {}`;

exports[`Test Home should UserView render 1`] = `ShallowWrapper {}`;
